### PR TITLE
Add new checker useless-object-inheritance.

### PR DESCRIPTION
**Pull request by sushobhit27**
### Fixes / new features
- #2177

### CODE REVIEW

File: CONTRIBUTORS.txt [hunk 1]
```diff
@@ -189,6 +189,7 @@ contributors:
   Added new check 'comparison-with-callable'.
   Removed six package dependency.
   Added new check 'chained-comparison'.
+  Added new check 'useless-object-inheritance'.
```
Comment (PCManticore) [inline CONTRIBUTORS.txt]: At this point you can stop adding these :) Just add `many more` 🥇

File: doc/whatsnew/2.0.rst [hunk 2]
```diff
@@ -25,6 +25,18 @@ Summary -- Release highlights
 
 New checkers
 ============
+* A new check was added, ``useless-object-inheritance``.
+
+  This refactoring message is emitted when pylint detects that a class inherits from object,
+  which is redundant as in Python 3, every class implicitly inherits from object.
```
Comment (PCManticore) [inline doc/whatsnew/2.0.rst]: `where every class implicitly`

File: pylint/test/functional/useless_object_inheritance.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+# pylint: disable=inconsistent-mro
+import abc
+
+class A(object):    # [useless-object-inheritance]
+    pass
+
+class B:
+    pass
+
+class C(B, object): # [useless-object-inheritance]
```
Comment (PCManticore) [inline pylint/test/functional/useless_object_inheritance.py]: Can you add a test where we inherit from a class, say `A` and make sure that the inherited class does not emit this error? e.g.
```
class A(object):
    pass
class B(A): # should not emit
    pass
```

File: CONTRIBUTORS.txt (hunk 1 above)
Comment (sushobhit27) [reply CONTRIBUTORS.txt]: will do it from the next commit :)

File: doc/whatsnew/2.0.rst (hunk 2 above)
Comment (sushobhit27) [reply doc/whatsnew/2.0.rst]: "which is redundant as in Python 3, where every class implicitly inherits from object." seems not correct to me grammatically to me, native speaker can comment on this. :smile:

File: pylint/test/functional/useless_object_inheritance.py (hunk 3 above)
Comment (sushobhit27) [reply pylint/test/functional/useless_object_inheritance.py]: added this test case.

**Comment by sushobhit27** [comment]
@PCManticore I have tried to adapt existing test cases mostly by disabling new checker in them, but in some non-functional test cases, I had to change the code itself.

**Comment by sushobhit27** [comment]
@PCManticore any review comments on this or shall I go ahead and merge it.

**Comment by PCManticore** [comment]
Yes, I wanted to follow up with something but didn't have the chance yet. I want to be able to analyze Python 2 files with pylint, and this check will have to be aware of that so it only runs on non Python 2 mode. But it's going to take me some time before I land that feature so free to merge it in the meantime.

**Pull request merged**